cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/JsonMenu/JsonLiveViewer/index.tsx

```diff
@@ -0,0 +1,152 @@
+
+const styles = (theme): StyleRules => {
+  return {
+    jsonViewer: {
+      zIndex: theme.zIndex.drawer,
+      width: JSON_VIEWER_WIDTH,
+    },
+    jsonViewerPaper: {
+      width: JSON_VIEWER_WIDTH,
+      backgroundColor: theme.palette.white[50],
+    },
+    toolbar: {
+      minHeight: '48px',
+    },
+    mainMenu: {
+      borderTop: `1px solid ${theme.palette.grey['500']}`,
+      paddingTop: theme.Spacing(1),
+      paddingBottom: theme.Spacing(1),
+    },
+    jsonActionButtons: {
+      padding: '5px',
+      display: 'flex',
+    },
+    closeJSONViewerButon: {
+      marginLeft: 'auto',
+    },
+    jsonViewerTooltip: {
+      fontSize: '14px',
+      backgroundColor: theme.palette.grey[500],
+    },
+  };
+};
+
+const JsonLiveViewerView: React.FC<ICreateContext & WithStyles<typeof styles>> = ({
+  classes,
+  pluginName,
+  pluginType,
+  displayName,
+  emitAlerts,
+  emitErrors,
+  configurationGroups,
+  groupToInfo,
+  groupToWidgets,
+  widgetToInfo,
+  jsonView,
+  setJsonView,
+}) => {
+  const widgetJSONData = {
+    pluginName,
+    pluginType,
+    displayName,
+    emitAlerts,
+    emitErrors,
+    configurationGroups,
+    groupToInfo,
+    groupToWidgets,
+    widgetToInfo,
+  };
+  const JSONConfig = getJSONConfig(widgetJSONData);
+  const downloadDisabled = pluginName.length === 0 || pluginType.length === 0;
+  return (
+    <div>
+      <Drawer
+        open={jsonView}
+        variant="persistent"
+        className={classes.jsonViewer}
+        anchor="right"
+        disableEnforceFocus={true}
+        ModalProps={{
+          keepMounted: true,
+        }}
+        classes={{
+          paper: classes.jsonViewerPaper,
+        }}
+        data-cy="navbar-drawer"
+      >
+        <div className={classes.toolbar} />
+        <List component="nav" dense={true} className={classes.mainMenu}>
+          <div className={classes.jsonActionButtons}>
+            <Tooltip
+              classes={{
+                tooltip: classes.jsonViewerTooltip,
+              }}
+              title={
+                downloadDisabled
+                  ? 'Download is disabled until the required fields are filled in'
+                  : 'Download Plugin JSON'
+              }
+            >
+              <span>
+                <Button
+                  disabled={downloadDisabled}
+                  onClick={() => downloadPluginJSON(widgetJSONData)}
+                >
+                  <SaveAltIcon />
+                </Button>
+              </span>
+            </Tooltip>
+            <Tooltip
+              classes={{
+                tooltip: classes.jsonViewerTooltip,
+              }}
+              title="Close JSON View"
+            >
+              <Button className={classes.closeJSONViewerButon} onClick={() => setJsonView(false)}>
+                <FullscreenExitIcon />
+              </Button>
+            </Tooltip>
+          </div>
+          <JsonEditorWidget
+            rows={50}
```

Comment:
Could this be static or is it just a placeholder for now? What happens on different browser heights?

Reply:
I set it as a placeholder, but I will use the code editor component directly and style that according to Edwin's suggestion!

Reply:
We ended up using `<pre>` element instead, so this won't be an issue anymore.

Reply:
Resolved!